src/course-unit/hooks.jsx

```diff
@@ -0,0 +1,74 @@
+import { useContext, useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { AppContext } from '@edx/frontend-platform/react';
+
+import { RequestStatus } from '../data/constants';
+import {
+  fetchCourseUnitQuery,
+  editCourseItemQuery,
+} from './data/thunk';
+import {
+  getCourseUnitData,
+  getLoadingStatus,
+  getSavingStatus,
+} from './data/selectors';
+import { updateSavingStatus } from './data/slice';
+import { getUnitViewLivePath, getUnitPreviewPath } from './utils';
+
+const useCourseUnit = ({ courseId, blockId }) => {
+  const dispatch = useDispatch();
+
+  const { config } = useContext(AppContext);
+  const courseUnit = useSelector(getCourseUnitData);
+  const savingStatus = useSelector(getSavingStatus);
+  const loadingStatus = useSelector(getLoadingStatus);
+
+  const [isTitleEditFormOpen, toggleTitleEditForm] = useState(false);
+
+  const unitTitle = courseUnit.metadata?.displayName || '';
+
+  const headerNavigationsActions = {
+    handleViewLive: () => {
+      window.open(config.LMS_BASE_URL + getUnitViewLivePath(courseId, blockId), '_blank');
+    },
+    handlePreview: () => {
+      const sectionId = courseUnit.ancestorInfo?.ancestors[0]?.id.split('@').pop();
+      window.open(config.PREVIEW_BASE_URL + getUnitPreviewPath(courseId, sectionId, blockId), '_blank');
+    },
```

Comment:
This function does not appear to be working for me. When I try to view the preview, I get a a page not found error
```
Page not found
No section found with name edx_introduction
Request Method: | GET
-- | --
http://preview.localhost:18000/courses/course-v1:edX+DemoX+Demo_Course/courseware/interactive_demonstrations/edx_introduction/1?activate_block_id=block-v1:edX+DemoX+Demo_Course+type@vertical+block@vertical_0270f6de40fc
lms.djangoapps.courseware.views.index.CoursewareIndex

```

Reply:
You're right, sorry for that.
I replaced the wrong `interactive_demonstrations` URL part with `sectionId`.